daemon/cmd/status.go

```diff
@@ -388,8 +388,21 @@ func (c *clusterNodesClient) NodeAdd(newNode nodeTypes.Node) error {
 
 func (c *clusterNodesClient) NodeUpdate(oldNode, newNode nodeTypes.Node) error {
 	c.Lock()
-	c.NodesAdded = append(c.NodesAdded, newNode.GetModel())
-	c.NodesRemoved = append(c.NodesRemoved, oldNode.GetModel())
+	// If the node is on the added list, just update it
+	found := -1
+	for i, added := range c.NodesAdded {
+		if added.Name == newNode.Fullname() {
+			found = i
+			break
+		}
+	}
+	if found != -1 {
+		c.NodesAdded[found] = newNode.GetModel()
```

Comment:
Couldn't this be moved to inside the loop and then the `found` variable wouldn't be needed?

Reply:
Since this i mostly a matter of style, I tried to follow the style of the function below (delete), but yes, I can move it inside the loop.